### ISSUE TITLE
[BUG] adding MPS GRUCell for efficiency

### ISSUE THREAD

**Issue by thegodone** [ml-explore/mlx#1500]
**Describe the bug**

This is related to the #1499 , the GRUCell is not implemented and the GRU version is also not optimized on MPS compare to pytorch and tensorflow variant.

**Comment by angeloskath** [comment]
I don't think this is related to #1499 but we definitely need to speed up our RNN implementations.

**Comment by zcbenz** [comment]
According to the discussion in https://github.com/ml-explore/mlx/pull/1508#issuecomment-2428282851 the the request for adding GRUCell was because it was claimed to accelerate speed, but the linked PyTorch code was not really a GPU accelerated kernel, it was just a C++ class gluing ops together so it won't run faster. Thus I'm closing this issue.
